go/private/rules/transition.bzl

```diff
@@ -191,6 +199,7 @@ _stdlib_keep_keys = sorted([
     "//go/config:pure",
     "//go/config:linkmode",
     "//go/config:tags",
+    "//go/config:pgoprofile",
```

Comment:
I'm not 100% sure I'm following this correctly but I think you need to add this to `_common_reset_transition_dict` as well, to prevent building depended-upon e.g. `cc_library` targets repeatedly for each profile input.

Reply:
@adam-azarchs makes sense. I've added that now

Reply:
Good catch, I will manually test that this works as expected before merging.

Reply:
This is working as expected.